File: solutions_5738606668808192_0/C++/IlyaKheifets/jj.cpp
#include <bits/stdc++.h>
using namespace std;

#define pow _pow

vector <int> res;
long long pow[11], num[11];
string a;

int main(){
	cout << "Case #1:\n";
	int cnt = 0;
	for (long long  i = (1ll << 15); i < (1ll << 16) && cnt < 50; ++i){
		if (!(i & 1)){
			continue;
		}
		for (int j = 2; j < 11; ++j){
			pow[j] = 1;
			num[j] = 0;
		}
		for (int k = 0; k < 16; ++k){
			for (int j = 2; j < 11; ++j){
				if (i & (1ll << k)){
					num[j] += pow[j];
				}
				pow[j] *= j;
			}
		}
		res.clear();
		for (int k = 2; k < 11; ++k){
			for (long long j = 2; j * j <= pow[k]; ++j){
				if (!(num[k] % j)){
					res.push_back(j);
					break;
				}
			}
		}
		if (res.size() == 9){
			cnt++;
			long long  l = i;
			a.clear(); 
			while (l){
				a += char('0' + (l & 1));
				l >>= 1;
			}
			reverse(a.begin(), a.end());
			cout << a;
			for (int k = 0; k < res.size(); ++k){
				cout << " " << res[k];
			}
			cout << "\n";
		}
	}
	return 0;
}

File: solutions_5738606668808192_1/Python/IlyaKheifets/jj.py
import sys

pw = [1] * 11
nm = [0] * 11
cnt = 0

print("Case #1:")
for i in range((1 << 31) + 1, 1 << 32, 2):
	if cnt == 500:
		break
	for j in range(2, 11):
		pw[j] = 1
		nm[j] = 0
	for k in range(32):
		for j in range(2, 11):
			if (i & (1 << k)) > 0:
				nm[j] += pw[j]
			pw[j] *= j
	res=[]
	for k in range(2, 11):
		t = True
		for j in range(2, 2016):
			if nm[k] % j == 0:
				res.append(j)
				t = False
				break
		if t:
			break
	if len(res) == 9:
		cnt += 1
		print("{0:b}".format(i), end = ' ')
		for r in range(len(res)):
			if r > 0:
				print(end = ' ')
			print(res[r], end = '')
		print()
